### DISCUSSION TITLE
New Development configuration where interpreter and XAML Hot Reload etc. are enabled, and Debug configuration is like Release

### DISCUSSION THREAD

**Discussion by janne-hmp** [dotnet/maui#21678] [Ideas]
.NET MAUI is currently using an interpreter for Debug build as default, which can potentially lead to large differences in the program behavior between Debug and Release configurations. Debug configuration should follow Release configuration as closely as possible, so that potential bugs in Release can be weeded out early. If the Debug configuration works very differently than Release, it sort of partly negates the point of the debug, at least for complicated errors arising from exact way the configuration is built, and as a way to ensure that Release configuration is stable if Debug is stable.

However, the goal of making the development quick via using interpreter, XAML Hot Reload etc. is understandable, but perhaps all such new features should be moved to a new configuration, which could be termed Development or something like that. Debug in turn should be as much like Release as possible.

**Comment by phillippschmedt** [comment]
@janne-hmp Ongoing discussion regarding the interpreter is here: https://github.com/dotnet/maui/issues/13019

**Comment by jonathanpeppers** [comment]
This idea is probably not very maintainable unless we had this feature in the .NET SDK:

* https://github.com/dotnet/sdk/issues/31918

We need a way to define a Debug-like configuration vs a Release-like configuration.

**Comment by janne-hmp** [comment]
There would be also a need for Publish configuration on iOS which would be like Release but signed with Apple Distribution provisioning profile, while Release would be signed with Apple Development provisioning profile to allow for testing on local devices.
